src/services/recommendations/recommendations.ts

```diff
@@ -0,0 +1,285 @@
+interface RecommendedCategory {
+  topic_id: string,
+  title: string,
+  score: number,
+  ranked_items: number[],
+}
+
+interface RankedList {
+  playlist_id: string
+  ranked_items: number[]
+}
+
+interface ContentRow {
+  slug: string,
+  name: string
+  ids: number[]
+}
+
+interface RankCategoriesResponse {
+  ranked_playlists: RankedList[]
+}
+
+interface RankedCategory {
+  slug: string
+  items: number[]
+}
+
+interface RankItemsResponse {
+  ranked_content_ids: number[]
+}
+
+interface Categories {
+  [key: string]: number[]
+}
+
+interface RecommendationsOptions {
+  section?: string
+}
+
+/**
+ * Fetches similar content to the provided content id
+ *
+ * @returns Returns the content_ids sorted by rank (most significant first)
+ * @example
+ * fetchSimilarItems(1113, 'drumeo')
+ *   .then(status => console.log(status))
+ *   .catch(error => console.error(error));
+ * @param content_id
+ * @param brand
+ * @param count
+ */
+export async function fetchSimilarItems(
+  content_id: number | string,
+  brand: string,
+  count: number = 10
+): Promise<number[] | null> {
+  if (!content_id) {
+    return []
+  }
+
+  const parsedContentId = typeof content_id === 'string' ? parseInt(content_id) : content_id
+
+  const data = {
+    brand: brand,
+    content_ids: parsedContentId,
+    num_similar: count + 1, // because the content itself is sometimes returned
+  }
+
+  const url = `/similar_items/`
+
+  try {
+    const httpClient = new HttpClient(baseURL)
+    const response = await httpClient.post<SimilarItemsResponse>(url, data)
+    // we requested count + 1 then filtered out the extra potential value, so we need slice to the correct size if necessary
+    return response.similar_items
+      .filter((item) => item !== parsedContentId)
+      .slice(0, count)
+  } catch (error) {
+    console.error('Fetch error:', error)
+    return null
+  }
+}
+
+/**
+ * Fetches recommended content from the PWRecsys
+ *
+ * @returns Returns the content_ids sorted by rank (most significant first)
+ * @example
+ * fetchSimilarItems('drumeo', 'song')
+ *   .then(status => console.log(status))
+ *   .catch(error => console.error(error));
+ * @param brand
+ * @param pageName
+ * @param count
+ */
+export async function fetchRecommendedCategories(
+  brand: string,
+  pageName: string,
+  count: number = 10
+): Promise<ContentRow[] | []> {
+  pageName = pageName.toLowerCase()
+  pageName = pageName.endsWith('s') ? pageName.slice(0, -1) : pageName;
+  if (pageName !== 'song' && pageName !== 'lesson') {
+    console.error("Invalid content type for", pageName)
+    return []
+  }
+
+  const data = {
+    brand: brand,
+    user_id: globalConfig.sessionConfig.userId,
+    content_type: pageName,
+    count: count,
+  }
+  const url = `/recommend_topics/`
+
+  try {
+    // TODO Remove before production deployment
+    //const httpClient = new HttpClient(tempBaseURL)
+    const httpClient = new HttpClient('https://MusoraProductDepartment-PWGeneratorDEV.hf.space')
```

Comment:
Blocking comment before production to update target
```suggestion
    const httpClient = new HttpClient(tempBaseURL)
```